src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -1362,6 +1362,27 @@ namespace {
 
     ASSERT_TRUE(!check_vector_eq(activeColumns, activeColumnsGlobal));
     ASSERT_TRUE(check_vector_eq(activeColumns, activeColumnsLocal));
+
+    // Assert that only columns with overlap are activated
+    numColumns = 16;
```

Comment:
Does these additions require the setup previously? Can you move these new checks into a stand alone named `TEST`?